pyomo/contrib/mindtpy/single_tree.py

```diff
@@ -657,256 +686,162 @@ def handle_lazy_subproblem_other_termination(
-        """This is an inherent function in LazyConstraintCallback in cplex.
+        """This is an inherent function in LazyConstraintCallback in CPLEX.
 
         This function is called whenever an integer solution is found during the branch and bound process.
         """
-        solve_data = self.solve_data
+        mindtpy_object = self.mindtpy_object
         config = self.config
         opt = self.opt
         main_mip = self.main_mip
+        mindtpy_object = self.mindtpy_object
 
-        if solve_data.should_terminate:
-            self.abort()
-            return
+        config.logger.debug(
+            "Solution source: %s (111 node_solution, 117 heuristic_solution, 119 mipstart_solution)".format(
+                self.get_solution_source()
+            )
+        )
 
-        self.handle_lazy_main_feasible_solution(main_mip, solve_data, config, opt)
+        # The solution found in MIP start process might be revisited in branch and bound.
+        # Lazy constraints separated when processing a MIP start will be discarded after that MIP start has been processed.
+        # This means that the callback may have to separate the same constraint again for the next MIP start or for a solution that is found later in the solution process.
+        # https://www.ibm.com/docs/en/icos/12.8.0.0?topic=SSSA5P_12.8.0/ilog.odms.cplex.help/refpythoncplex/html/cplex.callbacks.LazyConstraintCallback-class.htm
+        if (
+            self.get_solution_source() != 119
+            and mindtpy_object.mip_start_lazy_oa_cuts != []
+        ):
+            for constraint, sense, rhs in mindtpy_object.mip_start_lazy_oa_cuts:
+                self.add(constraint, sense, rhs)
+            mindtpy_object.mip_start_lazy_oa_cuts == []
 
+        if mindtpy_object.should_terminate:
+            self.abort()
+            return
+        self.handle_lazy_main_feasible_solution(main_mip, mindtpy_object, config, opt)
         if config.add_cuts_at_incumbent:
             self.copy_lazy_var_list_values(
                 opt,
                 main_mip.MindtPy_utils.variable_list,
-                solve_data.mip.MindtPy_utils.variable_list,
+                mindtpy_object.mip.MindtPy_utils.variable_list,
                 config,
             )
             if config.strategy == 'OA':
-                self.add_lazy_oa_cuts(solve_data.mip, None, solve_data, config, opt)
+                # The solution obtained from mip start might be infeasible and even introduce math domain error, like log(-1).
+                try:
+                    self.add_lazy_oa_cuts(
+                        mindtpy_object.mip, None, mindtpy_object, config, opt
+                    )
+                except ValueError:
+                    config.logger.debug(
+                        "The solution from MIP start is valid for the problem. "
+                        "Usually introduce a math domain error. "
+                        "We will abort it."
+                    )
```

Comment:
It would also be helpful to your users to append the error message from the ValueError to this message--right now you are swallowing it, hence losing some information about what went wrong.

Reply:
I agree with you. I'll append the error message here.